pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -324,17 +324,20 @@ type ScorePlugin interface {
 	ScoreExtensions() ScoreExtensions
 }
 
-// ReservePlugin is an interface for Reserve plugins. These plugins are called
-// at the reservation point. These are meant to update the state of the plugin.
-// This concept used to be called 'assume' in the original scheduler.
-// These plugins should return only Success or Error in Status.code. However,
-// the scheduler accepts other valid codes as well. Anything other than Success
-// will lead to rejection of the pod.
-type ReservePlugin interface {
+// ReservationPlugin is an interface for plugins with Reserve and Unreserve
+// methods. These are meant to update the state of the plugin. This concept
+// used to be called 'assume' in the original scheduler. These plugins should
+// return only Success or Error in Status.code. However, the scheduler accepts
+// other valid codes as well. Anything other than Success will lead to
+// rejection of the pod.
+type ReservationPlugin interface {
 	Plugin
 	// Reserve is called by the scheduling framework when the scheduler cache is
 	// updated.
 	Reserve(ctx context.Context, state *CycleState, p *v1.Pod, nodeName string) *Status
+	// Unreserve is called by the scheduling framework when the scheduler cache is
```

Comment:
`Unreserve is called by the scheduling framework when a reserved pod was rejected or an error occurred during reservation of other plugins or in a later phase. Unreserve must be idempotent.`

Reply:
Can you please elaborate why plugin authors must guarantee the idempotency of Unreserve operations? AFAIK there is no code path that will call the Unreserve method of a plugin more than once. In that case, whether or not a plugin's Unreserve is idempotent is immaterial. Most unreserves will likely be idempotent, but not all have to be (and it doesn't matter to us whether or not they are).

Reply:
It will depend on how you choose to implement #83557

A simple approach I was thinking of is just calling RunReservePluginsUnreserve. But to do that, Unreserve calls need to be idem-potent.

On the other hand, you could just call Unreserve directly in RunReservePluginsReserve, where you know which plugins succeded to run. But then there is the question: should we call Unreserve for the plugin that just failed?

In any case, we can leave this discussion for the follow up PR.

Reply:
Agreed, let's leave the discussion of idempotency to the other PR. (FWIW, since we know the index of the first reserve plugin that failed, we should be able to call Unreserve on just the other plugins that were previously reserved; whether or not the Unreserve should be called on the plugin that failed is a good question, though.)